gateway/src/proxy.rs

```diff
@@ -70,14 +67,19 @@ where
 }
 
 #[derive(Clone)]
-pub struct UserProxy {
-    gateway: Arc<GatewayService>,
-    task_sender: Sender<BoxedTask>,
+pub struct UserProxy<D>
+where
+    D: Dal + Send + Sync + 'static,
+{
+    dal: D,
     remote_addr: SocketAddr,
     public: FQDN,
 }
 
-impl<'r> AsResponderTo<&'r AddrStream> for UserProxy {
+impl<'r, D> AsResponderTo<&'r AddrStream> for UserProxy<D>
+where
+    D: Dal + Send + Sync + Clone + 'static,
```

Comment:
```suggestion
    D: Dal + 'static,
```
They are already on the Dal

Reply:
Woops, forgot about the proxy :see_no_evil: Fixed now